lib/DiscoveryStreamFeed.jsm

```diff
@@ -558,27 +598,70 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     // We can have no data if spocs set to off.
     // We can have no data if request fails and there is no good cache.
     // We want to send an update spocs or not, so client can render something.
-    spocs =
-      spocs && spocs.data
-        ? spocs
+    spocsState =
+      spocsState && spocsState.spocs
+        ? spocsState
         : {
             lastUpdated: Date.now(),
-            data: {},
+            spocs: {},
           };
 
-    let { data, filtered: frequencyCapped } = this.frequencyCapSpocs(
-      spocs.data
-    );
-    let { data: newSpocs, filtered } = this.transform(data);
+    let frequencyCapped = [];
+    let blockedItems = [];
+    let belowMinScore = [];
+    let campaignDupes = [];
+    this.placementsForEach(placement => {
+      const freshSpocs = spocsState.spocs[placement.name];
+
+      if (!freshSpocs || !freshSpocs.length) {
+        return;
+      }
+
+      const { data: capResult, filtered: caps } = this.frequencyCapSpocs(
+        freshSpocs
+      );
+      frequencyCapped = [...frequencyCapped, ...caps];
+
+      const { data: blockedResults, filtered: blocks } = this.filterBlocked(
```

Comment:
Trying with topsites front-end patch, filterBlocked is failing to filter dismissed spocs even when they are making browser.newtabpage.blocked pref